batch_size for multi_get

FewKinG opened the pull request:
Hey there,

I found it nice to have a :batch_size option also for the multi_get method.
If you are working with large rows with many columns or super columns you might get into trouble fetching those with multi_get.

The :batch_size option sets a limit to how many columns are fetched per key in a single query and like the get_range_batch method loops until all possible columns or at least :count columns have been optained.

However, due to the fact, that always up to :batch_size columns are fetched per query and key, you might get back up to :batch_size \* num_keys - 1 more columns than you specified with :count.

I've done only minimal testing to this feature and cannot guarantee that it always works as it should. But I want it to share, so others might use it as well.

Greetings,
FewKinG.

FewKinG committed:
Added option :batch_size to multi_get and implemented multi_get_batch

rwjblue commented:
This looks conceptually similar to get_range, get_range_single, and get_range_batch.  Would it be possible to structure it the same way?  So that a call to multi_get would either call multi_get_single or multi_get_batch based on whether or not a :batch_size was passed. That would keep the external API basically the same (with an extra option), and still allow a lot of flexibility.

Could you add some test coverage for the usual scenarios, update the CHANGELOG, and add information to the README?

FewKinG commented:
Yeah I'll structure it the same way. Basically that will just require some of the code in multi_get being transfered to a multi_get_single method as the multi_get method already allows the :batch_size option and calls the multi_get_batch in the proper case.

I will also write tests for it as soon as I have some time for that, maybe this evening.

FewKinG committed:
thrift_client dependency changed to 0.7.1

FewKinG committed:
implemented multi_get_batch and options for multi_get
Options are :batch_size, :keys_at_once and :type (the latter
is recommended to use if possible to ensure proper operation.
It defines how the columns are compared)

FewKinG committed:
Fixed multi_get_batch failing when needing keys_at_once and batch_size
together

FewKinG committed:
multi_get

Commit:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	cassandra.gemspec

Commit:
Bump version

FewKinG committed:
Fixed bug on keys_at_once option when using multi_get with standard
column families

FewKinG committed:
Merge branch 'master' of github.com:FewKinG/cassandra

FewKinG committed:
Fixed bug when passing inexistent keys to multi_get_batch

FewKinG committed:
Fixed typo

FewKinG committed:
version

Commit:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	cassandra.gemspec
	lib/cassandra/cassandra.rb

Commit:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	cassandra.gemspec

psanford commented:
Closing due to lack of interest. We can reopen if someone wants to clean this up.
